### ISSUE TITLE
[NoGame] HMM not detecting Sonic Generations ever since update

### ISSUE THREAD

**Issue by EFLeal** [thesupersonic16/HedgeModManager#602]
Updated when prompted, has been throwing this error ever since. Erased all the files related to HMM, reinstalled Sonic Generations, redownloaded the latest stable and dev versions, run in and out of the folder as admin, nothing.

[HMM_Snapshot_00032023133740.txt](https://github.com/thesupersonic16/HedgeModManager/files/11064964/HMM_Snapshot_00032023133740.txt)

HMM Info:
```
    Version: 7.9-0-cea68f9
    Args: 
    StartDir: C:\Program Files (x86)\Steam\steamapps\common\Sonic Generations\
    Process Level: Administrator
    Game: No Game Detected!
    Update Status: Up To Date
```

Exception:
```
    Type: ArgumentNullException
    Message: Value cannot be null.
Parameter name: source
    Source: System.Core
    Function: TSource FirstOrDefault[TSource](System.Collections.Generic.IEnumerable`1[TSource], System.Func`2[TSource,System.Boolean])
    StackTrace: 
       at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
       at HedgeModManager.Epic.SearchForGamesHeroic() in D:\a\HedgeModManager\HedgeModManager\HedgeModManager\Epic.cs:line 89
       at HedgeModManager.Epic.SearchForGames() in D:\a\HedgeModManager\HedgeModManager\HedgeModManager\Epic.cs:line 43
       at HedgeModManager.GameInstall.SearchForGames(String preference) in D:\a\HedgeModManager\HedgeModManager\HedgeModManager\Game.cs:line 275
       at HedgeModManager.HedgeApp.Main(String[] args) in D:\a\HedgeModManager\HedgeModManager\HedgeModManager\HedgeApp.xaml.cs:line 217
    InnerException: 
```

**Comment by thesupersonic16** [comment]
It appears to be having issues looking for your Legendary/Heroic games. Can you look for your `installed.json` file which for Heroic should be located in `%USERPROFILE%/.var/app/com.heroicgameslauncher.hgl/config/legendary`?

**Comment by EFLeal** [comment]
I had never even heard of that launcher until today, so the folder doesn't exist. I own the game on Steam, not sure why it's getting stuck on something that isn't even there

**Comment by EFLeal** [comment]
Managed to fix it. Tried installing Heroic to see if it would fix the error in the latest release (7.9-0-4e0282c), but it still threw the error. Thought it might have something to do with the dev build, so downloaded last stable build and ran it, and it went through. Opened the dev build after and it also went through. I'm thinking something went awry in the stable build which carried over into the dev build until I fixed it back in the stable build, but that's just conjecture on my part
